storyCite.lua

```diff
@@ -0,0 +1,248 @@
+local util_vars = require('Module:VarsUtil') --from https://thealchemistcode.fandom.com/wiki/Module:VarsUtil
+local util_link = require('Module:LinkUtil')
+local cache = require('mw.ext.LuaCache')
+local PREFIX = 'SC_CACHE_01-'
+local dump
+local p = {} --p stands for package
+local h = {} --h stands for helper
+
+-- from https://stackoverflow.com/a/7615129
+function h.splitString (inputstr, sep)
+		if sep == nil then
+				sep = "%s"
+		end
+		local t = {}
+		for str in string.gmatch(inputstr, "([^"..sep.."]+)") do
+				table.insert(t, str)
+		end
+		return t
+end
+
+function p.infoboxStore( frame )
+	local args = frame:getParent().args
+	local outputText = ""
+	if args["citation text"] then
+		outputText = args["citation text"]
+	else
+		if not args["separator"] then
+			args["separator"] = ","
+		end
+		
+		-- combines {{{anthology}}} and {{{audio anthology}}}
+		if args["anthology"] and args["audio anthology"] then
+			args["anthology"] = args["anthology"] .. ", " .. args["audio anthology"]
+		elseif (not args["anthology"]) and args["audio anthology"] then
+			args["anthology"] = args["audio anthology"]
+		end
+		
+		-- combines {{{adapted from}}} and {{{novelisation of}}}
+		if args["adapted from"] and args["novelisation of"] then
+			args["adapted from"] = args["adapted from"] .. ", " .. args["novelisation of"]
+		elseif (not args["adapted from"]) and args["novelisation of"] then
+			args["adapted from"] = args["novelisation of"]
+		end
+		
+		--gets and formats data for items that are unlinked	in the infobox
+		local data = {["anthology"] = "", ["writer"] = "", ["publisher"] = "", ["adapted from"] = ""}
+		for key, val in pairs(data) do
+			if args[key] then
+				local current = h.splitString(args[key], args["separator"])
+				for i = 1, #current do
+					data[key] = data[key] .. "[[" .. util_link.stripPipe(current[i]) .. "|" .. util_link.stripPipe(util_link.stripDab(current[i])) .. "]]"
+					if i == #current-1 then
+						data[key] = data[key] .. " and "
+					elseif i == #current then
+						data[key] = data[key]
+					else
+						data[key] = data[key] .. ", "
+					end
+				end
+			end
+		end
+		
+		--same concept as above but doesn't add links
```

Comment:
h.getAndFormatUnlinkedItems

then util_table.merge() the results of that into your data from the previous function

(see my TableUtil for `util_table.merge()`)